### PR TITLE
Add unit test for handling NSException with nil reason

### PR DESCRIPTION
**Pull request by ilyashumihin**
Fix possible crash 'plcrash_log_writer_set_exception' method when NSException instances have a nil reason.

AB#74114

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # PLCrashReporter Change Log
 
-## Version 1.4
+## Version 1.4.1 (Under development)
+
+* Fix possible crash `plcrash_log_writer_set_exception` method when NSException instances have a nil reason
```
Comment (MatkovIvan) [inline CHANGELOG.md]:
```suggestion
* Fix possible crash `plcrash_log_writer_set_exception` method when `NSException` instances have a `nil` reason.

___
```

File: Source/PLCrashLogWriter.m [hunk 2]
```diff
@@ -476,7 +476,7 @@ void plcrash_log_writer_set_exception (plcrash_log_writer_t *writer, NSException
     /* Save the exception data */
     writer->uncaught_exception.has_exception = true;
     writer->uncaught_exception.name = strdup([[exception name] UTF8String]);
-    writer->uncaught_exception.reason = strdup([[exception reason] UTF8String]);
+    writer->uncaught_exception.reason = strdup([exception reason] != nil ? [[exception reason] UTF8String] : "");
```
Comment (MatkovIvan) [inline Source/PLCrashLogWriter.m]: Please pick up this change from master because #21 is merged.

**Comment by ilyashumihin** [comment]
@clpolet Yes, we merged the changes from the author's branch and covered them with tests.
